library/src/main/java/org/apache/apex/malhar/lib/window/Window.java

```diff
@@ -148,20 +136,49 @@ public boolean equals(Object other)
       }
     }
 
+    @Override
+    public int hashCode()
+    {
+      int result = (int)(beginTimestamp ^ (beginTimestamp >>> 32));
+      result = 31 * result + (int)(durationMillis ^ (durationMillis >>> 32));
+      return result;
+    }
+
+    @Override
+    public int compareTo(TimeWindow o)
+    {
+      if (this.getBeginTimestamp() < o.getBeginTimestamp()) {
+        return -1;
+      } else if (this.getBeginTimestamp() > o.getBeginTimestamp()) {
+        return 1;
+      } else if (this.getDurationMillis() < o.getDurationMillis()) {
+        return -1;
+      } else if (this.getDurationMillis() > o.getDurationMillis()) {
+        return 1;
+      }
+      return 0;
+    }
+
+    @Override
+    public String toString()
+    {
+      return "[TimeWindow " + getBeginTimestamp() + "(" + getDurationMillis() + ")]";
```

Comment:
WindowOption.SlidingTimeWindows generates Window.TimeWindow itself. There is no class that represents individual sliding time windows.